validator/client/src/main/java/tech/pegasys/teku/validator/client/BlockDutyScheduler.java

```diff
@@ -46,4 +49,14 @@ public void onBlockProductionDue(final UInt64 slot) {
 
     notifyEpochDuties(EpochDuties::onBlockProductionDue, slot);
   }
+
+  @Override
+  protected Bytes32 getExpectedTargetRoot(
+      final Bytes32 headBlockRoot,
+      final Bytes32 previousTargetRoot,
+      final Bytes32 currentTargetRoot,
+      final UInt64 headEpoch,
+      final UInt64 dutyEpoch) {
+    return headEpoch.equals(dutyEpoch) ? currentTargetRoot : headBlockRoot;
```

Comment:
Looks like these methods assume that dutyEpoch >= headEpoch, maybe we should assert that here to be safer / clearer.

Reply:
Yep, it's called after doing a tailMap so always guaranteed to meet that condition - checkArgument added.